linker.py

```diff
@@ -166,6 +169,7 @@ def get_symlink_pairs(ref, *, det_map, root_map=None):
                             det_name=det_name,
                             N=point_number * fpp + fr,
                             det_type=det_type,
+                            sample=sample_name,
```

Comment:
```suggestion
                            sample=doc['data'].get('target_file_name', sample_name),
```
You might do this in one line and remove the code above.